# Sensor inputs can be of type
# init (initialize the filter with this odometry measurement)
# imu (update the filter with an IMU measurement)
# odom (update the filter with an odom measurement)
# groundtruth (Compare the current state against a groundtruth measurement)

# Measurement covariance can be provided here, or if the measurement noise is
# specified in the message itself/not used for the message type, replaced with None

sensor_inputs:
  - topic: "odom"
    type: "init"
    meas_covar: None

  - topic: "odom"
    type: "groundtruth"
    meas_covar: None

  - topic: "imu"
    type: "imu"
    meas_covar: [[0.01, 0.0], [0.0, 0.01]]

  - topic: "odom"
    type: "init"
    meas_covar: None

  - topic: "odom_noisy"
    type: "odometry"
    meas_covar: [[0.1, 0.0], [0.0, 0.5]]

  - topic: "lidar_odom"
    type: "odometry"
    meas_covar: [[0.1, 0.0], [0.0, 0.1]]

  - topic: "visual_odom"
    type: "odometry"
    meas_covar: [[0.001, 0.0], [0.0, 0.001]]


# Provide initial state and covariance values here, along with the process noise
init_state: [0.,0.,0.,0.,0.,0.]
init_cov_sigmas: [.5, .5, 0.001, 0.001, 0.001, 0.001]
proc_noise_sigmas: [0.01, 0.01, 0.001, 0.25, 0.025, 0.0015]
